src/main.cpp

```diff
@@ -17,7 +17,8 @@
 #include <znc/znc.h>
 #include <signal.h>
 
-#if defined(HAVE_LIBSSL) && defined(HAVE_PTHREAD)
+#if defined(HAVE_LIBSSL) && defined(HAVE_PTHREAD) && \
+( !defined(OPENSSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER < 0x10100004 ) /* 1.1.0-pre4: openssl/openssl@2e52e7df518d80188c865ea3f7bb3526d14b0c08 */
```

Comment:
Ah, I understand.

I guess that should be made more clear. Some comment saying so explicitly so. Something like "Starting with version 1.1.0-pre4, OpenSSL automatically uses pthreads / the windows locking API without needing help. See commit  openssl/openssl@2e52e7df518d80188c865ea3f7bb3526d14b0c08."